### PR TITLE
Refactor dead sections

### PR DESCRIPTION
**Pull request by Castavo**
linked to: #4660 

In this PR there is, for now
* Removal of an unused field (planned to use it but it got scrapped)
* Renaming of classes and attributes

### CODE REVIEW

**Review by clarani — COMMENTED** [review]
Great PR, thank you :pray:
I can approve the frontend part once the PR is ready (I will have to test)

File: core/kt-osrd-sim-infra/src/main/kotlin/fr/sncf/osrd/sim_infra/api/TrackProperties.kt
Comment (multun) [inline core/kt-osrd-sim-infra/src/main/kotlin/fr/sncf/osrd/sim_infra/api/TrackProperties.kt]:
```suggestion
    fun getTrackChunkNeutralSections(trackChunk: DirTrackChunkId): DistanceRangeMap<NeutralSection>
```

File: core/kt-osrd-sim-infra/src/main/kotlin/fr/sncf/osrd/sim_infra/impl/RawInfraImpl.kt
Comment (multun) [inline core/kt-osrd-sim-infra/src/main/kotlin/fr/sncf/osrd/sim_infra/impl/RawInfraImpl.kt]:
```suggestion
    override fun getTrackChunkNeutralSections(trackChunk: DirTrackChunkId): DistanceRangeMap<NeutralSection> {
```

File: core/osrd-railjson/src/main/java/fr/sncf/osrd/railjson/schema/infra/RJSInfra.java
Comment (multun) [inline core/osrd-railjson/src/main/java/fr/sncf/osrd/railjson/schema/infra/RJSInfra.java]: That's quite a bump, why is that? I'd rather have a single version bump and migration

File: editoast/src/tests/small_infra/small_infra.json
Comment (multun) [inline editoast/src/tests/small_infra/small_infra.json]: This will need a rebase once https://github.com/osrd-project/osrd/pull/4728 is merged

**Review by multun — CHANGES_REQUESTED** [review]

**Comment by multun** [comment]
A very nice PR, thanks :) Please change it as you see fit, and ask @flomonster to merge it you think it's ready (all core maintainers are away next week)

**Review by flomonster — APPROVED** [review]
Thanks for this PR!
LGTM

**Review by clarani — APPROVED** [review]
Front looks good to me, tested in local :heavy_check_mark:

**Review by younesschrifi — APPROVED** [review]
LGTM

**Pull request merged**
